### PR TITLE
Guide for managing domains

### PR DESCRIPTION
**Pull request by marians**
WIP

The service isn't deployed yet, the web UI isn't there yet, and the ingress controller isn't either, but I figured I could start iterating on the docs.

@puja108 RFR
@JosephSalisbury FYI
## TODO
- [x] Add part to ingress guide about custom domains, once https://github.com/giantswarm/docs-content/pull/212 is merged
- [ ] Merge once Ingress controller supports custom domains

### CODE REVIEW

**Comment by marians** [comment]
@puja108 @oponder I updated the guide content. Feel free to (re-)check.

File: content/guides/managing-domains/index.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+Using the organization selector, open the details page for that organization.
+
+Find the *Domains* section on the page and click the *Add Domain* button.
+
+You'll now have to decide which domain to add.
+In order to use the domain `api.messaging.example.com`, you can either add exactly that name, or you can add one of it's superior domains: `messaging.example.com` or even `example.com`.
+Adding a superior domain will give you the ability to use all subdomains of the added domain, including the added domain itself, within your organization's clusters, without having to repeat this step for every subdomain.
+
+As you'll be asked to add a TXT record for the domain you add, this might affect your decision, too. For example, you might now want to have the TXT record in the entry `example.com`, but are fine with adding it to `messaging.example.com`. The decision is up to you, really.
```
Comment (puja108) [inline content/guides/managing-domains/index.md]: `now`-> `not`

File: content/guides/managing-domains/index.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+- The DNS server update might take between a moment and several hours, depending on the provider you use.
+- Our bots take a break between checks, usually up to a minute.
+- If your domain hasn't been validated yet, you'll find some information in the domains list of our web user interface, just where you added that domain.
+- We will only look for up to one day (or 1000 checks, whatever is reached first). After that, the domain will be marked as *Failed*. In this case, you can delete the domain and add it again to try your luck.
+- If things don't work out, please contact our support.
+
+
+## Further reading
+
+Once you have a domain validated, you might want to find out how to use it with your services running on Kubernetes. We have a guide for that: [Accessing Services from the Outside](/guides/accessing-services-from-the-outside/). Pay special attention to the section called [Setting up a Public Ingress](/guides/accessing-services-from-the-outside/#setting-up-a-public-ingress).
```
Comment (marians) [inline content/guides/managing-domains/index.md]: Good point. I'll try my luck with the help of @JosephSalisbury

File: content/guides/managing-domains/index.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+Our diligent bots will pick up the DNS change soon after it's done. Once they find the expected validation token in your domain's TXT records, the domain is validated for the organization and can be used within all of the organization's clusters.
+
+Once validated, you find the domain's status displayed in the web user interface where you added it before.
+
+A few notes:
+
+- The DNS server update might take between a moment and several hours, depending on the provider you use.
+- Our bots take a break between checks, usually up to a minute.
+- If your domain hasn't been validated yet, you'll find some information in the domains list of our web user interface, just where you added that domain.
+- We will only look for up to one day (or 1000 checks, whatever is reached first). After that, the domain will be marked as *Failed*. In this case, you can delete the domain and add it again to try your luck.
```
Comment (marians) [inline content/guides/managing-domains/index.md]: Yes

File: content/guides/managing-domains/index.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+# Managing Domains
+
+When exposing a service running on one of your Giant Swarm Kubernetes clusters to the outside world, you will most likely want to use your preferred domain name.
+This guide explains how to make domains available to the system before you can use them in ingresses.
+
+In general, the following steps are involved:
+
+1. Setting up a CNAME
+2. Verifying your domain ownership
+3. Setting up ingress
```
Comment (JosephSalisbury) [inline content/guides/managing-domains/index.md]: Should be `Ingress`

**Comment by marians** [comment]
@puja108 Custom domain example is added to the ingress guide. RFR

**Comment by puja108** [comment]
LGTM

File: content/guides/managing-domains/index.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+Note that it can take some time until the DNS propagates a change for a domain name.
+So if a record for `api.messaging.example.com` existed before, the change might not appear immediately.
+The duration is subject to configuration of your DNS records.
+
+## Verifying your domain ownership {#ownership}
+
+This step ensures that only your clusters (or, to be more specific: the clusters of your organizations) may have services responding to requests for your domain.
+
+### Step 1: Add the domain to your organization's domains {#adding}
```
Comment (oponder) [inline content/guides/managing-domains/index.md]: How about just "Add the domain to your organization" ?

File: content/guides/managing-domains/index.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+Log in to our [web user interface](https://happa.giantswarm.io/) and make sure you have the right organization selected.
+
+Using the organization selector, open the details page for that organization.
+
+Find the *Domains* section on the page and click the *Add Domain* button.
+
+You'll now have to decide which domain to add.
+In order to use the domain `api.messaging.example.com`, you can either add exactly that name, or you can add one of it's superior domains: `messaging.example.com` or even `example.com`.
```
Comment (oponder) [inline content/guides/managing-domains/index.md]: `it's` should be `its` in this case

File: content/guides/managing-domains/index.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+A few notes:
+
+- The DNS server update might take between a moment and several hours, depending on the provider you use.
+- Our bots take a break between checks, usually up to a minute.
+- If your domain hasn't been validated yet, you'll find some information in the domains list of our web user interface, just where you added that domain.
+- We will only look for up to one day (or 1000 checks, whatever is reached first). After that, the domain will be marked as *Failed*. In this case, you can delete the domain and add it again to try your luck.
+- If you try to add a domain again which has already been added before by your organization and *Failed* the validation, the *Failed* entry will be deleted and the domain will be added again. Validation attempts start again.
+- If you already have a *Failed* domain entry like `subsub.sub.example.com` and then try to add the superior `sub.example.com`, the entry for `subsub.sub.example.com` will be deleted, as the superior entry will cover all sub domains.
+- If things don't work out, please contact us for support.
```
Comment (oponder) [inline content/guides/managing-domains/index.md]: Nice to always include where to go for support, or a link to a page that has more info when mentioning support?

Comment (marians) [inline content/guides/managing-domains/index.md]: I'll leave this for a general footer.

**Comment by webwurst** [comment]
Can this be closed?

**Pull request closed without merging**